storage/rocksdb/ha_rocksdb.cc

```diff
@@ -6327,17 +6326,22 @@ int ha_rocksdb::read_row_from_secondary_key(uchar *const buf,
   int rc = 0;
   uint pk_size;
 
-  if (m_keyread_only && m_lock_rows == RDB_LOCK_NONE && !has_hidden_pk(table)) {
-    /* Get the key columns and primary key value */
-    const rocksdb::Slice &rkey = m_scan_it->key();
+  /* Get the key columns and primary key value */
+  const rocksdb::Slice &rkey = m_scan_it->key();
+  const rocksdb::Slice &value = m_scan_it->value();
+
+  bool covered_lookup = m_keyread_only || kd.covers_lookup(table, &value);
```

Comment:
Do we know how much this affects performance of selects via secondary indexes? Could we measure this using sysbench.